services/LayersService.js

```diff
@@ -175,3 +150,47 @@ export default class LayersService {
     });
   }
 }
+
+/**
+ * Fetches a layer according to widget id and params.
+ *
+ * @param {String} id - layer id.
+ * @param {Object[]} params - params sent to the API.
+ * @returns {Object[]} - serialized specific layer.
+ */
+
+export const fetchLayer = (id, params = {}) => {
+  if (!id) throw Error('layer id is mandatory to perform this fetching.');
+  logger.info(`'Fetching layer id (${id}) '`);
+  return WRIAPI.get(`/layer/${id}`, {
+    headers: {
+      ...WRIAPI.defaults.headers,
+      // TO-DO: forces the API to not cache, this should be removed at some point
+      'Upgrade-Insecure-Requests': 1
+    },
+    params: {
+      application: process.env.APPLICATIONS,
+      ...params
+    },
+    transformResponse: [].concat(
+      WRIAPI.defaults.transformResponse,
+      ({ data }) => data
+    )
+  }).then((response) => {
+    const { status, statusText, data } = response;
+    logger.debug(`Widget layer fetch returned with code ${status}`);
```

Comment:
Do not debug this, only possible errors of beginning of actions, remember me tomorrow to tell you more about this.